### PR TITLE
create rule fixed by sending categoryId, added name filters at

### PR DESCRIPTION
**Pull request by theibrahim**
Frontend: Fixed create rule issue by sending categoryId instead of category

Backend: Added name filter support

### CODE REVIEW

**Review by pauloouriques — APPROVED** [review]
looks good

File: components/scoped/CreateNewRule.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ const RePricer = ({ onBack }) => {
     setIsLoading(true);
     const data = {
       name: name,
-      category: category,
+      categoryId: category,
```
Comment (bali710) [inline components/scoped/CreateNewRule.js:28]: this part is already handled

File: pages/api/pricing-rules/list.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ const handler = async (req, res) => {
   const business = await prisma.business.findUnique({
     where: { email: session.user.email },
   });
-  const { categoryId, listingType } = req.query;
+  const { categoryId, listingType, name } = req.query;
```
Comment (bali710) [inline pages/api/pricing-rules/list.js:16]: this one handled as well

File: pages/api/pricing-rules/list.js [hunk 3]
```diff
@@ -24,6 +24,10 @@ const handler = async (req, res) => {
       if (listingType) {
         condition = { ...condition, listingType };
       }
+      if (name) {
```
Comment (bali710) [inline pages/api/pricing-rules/list.js:27]: this part also handled

**Review by bali710 — CHANGES_REQUESTED** [review]
changes related to the filter has been handled

**Pull request closed without merging**